infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogEntryWriter.java

```diff
@@ -177,5 +187,9 @@ public boolean apply(LogReplicationEntryMsg msg) throws ReplicationWriterExcepti
     public void reset(long snapshot, long ackTimestamp) {
         srcGlobalSnapshot = snapshot;
         lastMsgTs = ackTimestamp;
+        if (config.getStreamsDiscoveryMode().equals(StreamsDiscoveryMode.DYNAMIC)) {
+            // If LR is in DYNAMIC mode, sync with registry table to avoid data loss
+            config.syncWithRegistry();
```

Comment:
reset() is invoked during initialization in 
https://github.com/CorfuDB/CorfuDB/blob/master/infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationSinkManager.java#L203

Additionally, the constructor for this class also does a sync with the registry table through the SinkWriter just before reset() is invoked.

You can move the logic for syncing outside this method and call it explicitly wherever required.